Include/cpython/code.h

```diff
@@ -19,21 +19,35 @@ extern "C" {
 typedef union {
     uint16_t cache;
     struct {
-        uint8_t opcode;
-        uint8_t oparg;
-    };
+        uint8_t code;
+        uint8_t arg;
+    } op;
 } _Py_CODEUNIT;
 
-#define _Py_OPCODE(word) ((word).opcode)
-#define _Py_OPARG(word) ((word).oparg)
+
+/* These macros only remain defined for compatibility. */
+#define _Py_OPCODE(word) ((word).op.code)
+#define _Py_OPARG(word) ((word).op.arg)
+
+static inline uint16_t
+_py_make_codeunit(uint8_t opcode, uint8_t oparg)
+{
+    // No designated initialisers because of C++ compat
+    _Py_CODEUNIT word;
+    word.op.code = opcode;
+    word.op.arg = oparg;
+    return word.cache;
+}
 
 static inline void
-_py_set_opcode(_Py_CODEUNIT *word, uint8_t opcode)
+_py_set_opcode(_Py_CODEUNIT *word, uint16_t packed_opcode)
 {
-    word->opcode = opcode;
+    word->cache = packed_opcode;
```

Comment:
This appears to be wrong.
It should be `word->op.code = opcode`

Reply:
Yeah, you're right. I thought I had reimplemented the previous versions semantics properly, but turns out I got it wrong.